include/http/request_handler.js

```diff
@@ -534,15 +535,26 @@ module.exports = function RequestHandlerModule(pb) {
      */
     RequestHandler.prototype.handleRequest = function(){
 
-        //fist things first check for public resource
+        //first things first check for public resource
         if (RequestHandler.isPublicRoute(this.url.pathname)) {
             return this.servePublicContent();
         }
 
         //check for session cookie
         var cookies = RequestHandler.parseCookies(this.req);
+        var siteObj = RequestHandler.sites[this.hostname];
+        var url = this.url.pathname;
         this.req.headers[pb.SessionHandler.COOKIE_HEADER] = cookies;
 
+        if(typeof(siteObj) !== 'undefined' && siteObj.hasOwnProperty('forceLocale') && siteObj.forceLocale) {
+            cookies.locale = cookies.locale || cookies[" locale"];
+            cookies.locale = cookies.locale || siteObj.defaultLocale;
+            if (checkRouteContainsLocale(url, siteObj)) {
+                var redirectLocation = pb.SiteService.getHostWithProtocol(this.hostname);
+                redirectLocation += '/' + cookies.locale + this.url.path;
+                return this.doRedirect(redirectLocation, pb.HttpStatus.MOVED_TEMPORARILY);
+            }
+        }
```

Comment:
I think this is the wrong spot to do this.  If you wait until the route is derived then you can check the route spec to see if the route is localized instead of having to guess through your ```checkRouteContainsLocale```.  Your implementation assumes routes are localized if a supported locale exists within URL path.  This is an incorrect assumption.